Polish babel parser (#2413)
babel/website: docs/parser.md

| `typescript` ([repo](https://github.com/Microsoft/TypeScript)) | `var a: string = "";` |
| `v8intrinsic` | `%DebugPrint(foo);` |

> #### ECMAScript [proposals](https://github.com/tc39/proposals)

Reviewer: I'm okay with leaving this pointed to our proposals repo as it has additional Babel-related info.

@hzoo thoughts?

Reviewer 2: That's fine, but we haven't been keeping as much up to date as we could to bring more of that context to light (though it shouldn't be as hard to do so..)

Author: ```suggestion
#### ECMAScript [proposals](https://github.com/babel/proposals)
```
Reverted this change, we can discuss how to deal with our proposals repo later.

| `doExpressions` ([proposal](https://github.com/tc39/proposal-do-expressions)) | `var a = do { if (true) { 'hi'; } };` |
| `exportDefaultFrom` ([proposal](https://github.com/tc39/ecmascript-export-default-from)) | `export v from "mod"` |
| `functionBind` ([proposal](https://github.com/zenparsing/es-function-bind)) | `a::b`, `::console.log` |
| `importAssertions` ([proposal](https://github.com/tc39/proposal-import-assertions)) | `import json from "./foo.json" assert { type: "json" };` |
| `moduleStringNames` ([proposal](https://github.com/tc39/ecma262/pull/2154)) | `import { "😄" as smile } from "emoji";` |
| `partialApplication` ([proposal](https://github.com/babel/proposals/issues/32)) | `f(?, a)` |
| `pipelineOperator` ([proposal](https://github.com/babel/proposals/issues/29)) | <code>a &#124;> b</code> |
| `privateIn` ([proposal](https://github.com/tc39/proposal-private-fields-in-in)) | `#p in obj` |
| `throwExpressions` ([proposal](https://github.com/babel/proposals/issues/23)) | `() => throw new Error("")` |
| `topLevelAwait` ([proposal](https://github.com/tc39/proposal-top-level-await/)) | `await promise` in modules |

#### Latest ECMAScript features

> These features are already enabled on latest `@babel/parser`, they can not be disabled because they are part of the language.

Reviewer: ```suggestion
The following features are already enabled on the latest version of `@babel/parser`, and cannot be disabled because they are part of the language.
```
